ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,467 @@
+        # Pdb().set_trace()
+        offset = 0
+        if request.pageToken:
+            print("pageToken " + request.pageToken)
+            offset = request.pageToken.split(':')[1]
+            print("offset " + offset)
+
+        associations = self.query_labels(request.feature,
+                                         request.evidence,
+                                         request.phenotype,
+                                         request.pageSize,
+                                         offset
+                                         )
+        # from IPython.core.debugger import Pdb ;        Pdb().set_trace()
+
+        self.associations_length = len(associations)
+        for association in associations:
+            yield association
+
+    def query_labels(self, location=None, drug=None,
+                     disease=None, pageSize=None, offset=0):
+        """
+        Thus query is the main search mechanism.
+        It queries graph for annotations that match wild card search of
+        lables or predicate URI
+        """
+        query = self.format_query(location, drug, disease)
+
+        query = query.replace("%PROPERTIES%",
+                              "distinct ?s  ?location ?location_label " +
+                              "?disease ?disease_label ?drug  ?drug_label")
+
+        query += ("LIMIT %s OFFSET %s " % (pageSize, offset))
+
+        print(query)
+
+        print("starting query")
+        results = G2PDataset.g.query(query)
+        rows = [row.asdict() for row in results]
+        print("ending query len(rows)=%s" % len(rows))
+
+        annotations = []
+        for row in results:
+            annotations.append(
+                self.to_ga4gh(
+                              self.query("FILTER (?s = <" +
+                                         row['s'].toPython()+">)")))
+
+        return annotations
+
+    def query_counts(self, location=None, drug=None, disease=None):
+        """
+        Thus query is the main search mechanism.
+        It queries graph for annotations that match wild card search of
+        lables or predicate URI
+        """
+        query = self.format_query(location, drug, disease)
+        count_query = query.replace(
+            "%PROPERTIES%", "(COUNT( distinct ?s ) AS ?count) ")
+        print(count_query)
+        results = G2PDataset.g.query(count_query)
+        rows = [row.asdict() for row in results]
+        count = rows[0]['count'].toPython()
+
+        return count
+
+    def format_query(self, location=None, drug=None, disease=None):
+        """
+        Generate a formatted sparql query with appropriate filters
+        """
+        query = self.search_query
+        if (location is None and drug is None and disease is None):
+            raise exceptions.NotImplementedException(
+               "At least one of [location, drug, disease] must be specified")
+        filters = []
+
+        if (location and isinstance(location, basestring)):
```

Comment:
Always check for ``None`` explicitly. Plus, this is redundant, isn't it? I don't think None can be an instance of basestring.

How can location be either a string or a dict here? It's not obvious how this can happen reading the rest of the code.